Controllers/controller.py

```diff
@@ -106,3 +109,20 @@ def open_file_dialog(self):
             url = file_dialog.selectedUrls()[0]
             self._media_player.setSource(url)
             self._media_player.play()
+            
+    @Slot()
+    def play_video(self):
+        print("Clicked")
+        if self._media_player.playbackState() == QMediaPlayer.PlayingState:
+            self._media_player.pause()
+        else:
+            self._media_player.play()
+        self.mediastate_changed()
+
+    @Slot()
+    def mediastate_changed(self):
```

Comment:
Rename this function according to its purpose. It looks like it is just toggling the icon of the button. A better name might be "toggle_play_pause_icon". 

Also, make sure that your function definitions are in alphabetical order.